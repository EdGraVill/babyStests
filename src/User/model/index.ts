import { model } from 'mongoose';
import schema from './schema';
import addStatics from './addStatics';

addStatics(schema);

export default model<User.Document, User.Model>('User', schema);
