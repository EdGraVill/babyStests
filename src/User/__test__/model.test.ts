import { compare } from 'bcrypt';
import User from '../model';

describe('User model', (): void => {
  it('Must throw an error if required fields aren\'t provided', async (): Promise<void> => {
    const fakeUser = new User({});

    try {
      await fakeUser.validate();
    } catch (error) {
      expect(error).toBeInstanceOf(Error);
      expect(error).toMatchSnapshot();
    }
  });

  it('Must throw an error if curp isn\'t valid', async (): Promise<void> => {
    const fakeUser = new User({
      apellidos: 'Grajales Villanueva',
      curp: 'GAVE950315HVZRLD',
      email: 'edgravill@gmail.com',
      estadoNacimiento: 'Veracruz',
      fechaNacimiento: '1995-03-15',
      nombres: 'Eduardo',
      password: 'sVcfwa5W;<@6+xqp',
      sexo: 'hombre',
    });

    try {
      await fakeUser.validate();
    } catch (error) {
      expect(error).toBeInstanceOf(Error);
      expect(error.message).toBe(`User validation failed: curp: Path \`curp\` is invalid (${fakeUser.curp}).`);
    }
  });

  it('Must hash the password', async (): Promise<void> => {
    const fakePassword = 'sVcfwa5W;<@6+xqp';
    const fakeUser = new User({
      apellidos: 'Grajales Villanueva',
      curp: 'GAVE950315HVZRLD02',
      email: 'edgravill@gmail.com',
      estadoNacimiento: 'Veracruz',
      fechaNacimiento: '1995-03-15',
      nombres: 'Eduardo',
      password: fakePassword,
      sexo: 'hombre',
    });

    try {
      const result = await compare(fakePassword, fakeUser.password);

      expect(result).toBeTruthy();
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });

  it('Must be valid user', async (): Promise<void> => {
    const fakeUser = new User({
      apellidos: 'Grajales Villanueva',
      curp: 'GAVE950315HVZRLD02',
      email: 'edgravill@gmail.com',
      estadoNacimiento: 'Veracruz',
      fechaNacimiento: '1995-03-15',
      nombres: 'Eduardo',
      password: 'sVcfwa5W;<@6+xqp',
      sexo: 'hombre',
    });

    try {
      await fakeUser.validate();
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });
});
