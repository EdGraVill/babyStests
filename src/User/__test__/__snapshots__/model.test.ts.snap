// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User model Must throw an error if required fields aren't provided 1`] = `[ValidationError: User validation failed: sexo: Path \`sexo\` is required., password: Path \`password\` is required., nombres: Path \`nombres\` is required., fechaNacimiento: Path \`fechaNacimiento\` is required., estadoNacimiento: Path \`estadoNacimiento\` is required., email: Path \`email\` is required., curp: Path \`curp\` is required., apellidos: Path \`apellidos\` is required.]`;
